### PR TITLE
Add some more retry...

### PR DESCRIPTION
**Pull request by brendandburns**
The list op flaked in e2e recently.  Add sleep in a few other places, since I noticed it.

@zmerlynn @quinton-hoole @jlowdermilk

### CODE REVIEW

**Comment by zmerlynn** [comment]
LGTM

File: cluster/gce/list-resources.sh [hunk 1]
```diff
@@ -37,7 +37,17 @@ GREP_REGEX=""
 function gcloud-compute-list() {
   local -r resource=$1
   echo -e "\n\n[ ${resource} ]"
-  gcloud compute ${resource} list --project=${PROJECT} ${@:2} | grep "${GREP_REGEX}"
+  local attempt=1
+  local result=""
+  while true; do
```
Comment (quinton-hoole) [inline cluster/gce/list-resources.sh]: You'll need a max retires, or a timeout in this loop, otherwise it could go forever.

**Comment by quinton-hoole** [comment]
I think you're missing a timeout on that potentially endless loop? Removing LGTM for now.

**Comment by quinton-hoole** [comment]
Not sure what's happening in e2e here.  Looks like Jenkins gcloud installation might be borked?

```
All components are up to date.
14:06:04 Traceback (most recent call last):
14:06:04   File "/usr/local/bin/../share/google/google-cloud-sdk/./lib/googlecloudsdk/gcloud/gcloud.py", line 187, in main
14:06:04     _cli.Execute()
14:06:04   File "/usr/local/bin/../share/google/google-cloud-sdk/./lib/googlecloudsdk/calliope/cli.py", line 592, in Execute
14:06:04     result = args.cmd_func(cli=self, args=args)
14:06:04   File "/usr/local/bin/../share/google/google-cloud-sdk/./lib/googlecloudsdk/calliope/backend.py", line 1223, in Run
14:06:04     resources = command_instance.Run(args)
14:06:04   File "/usr/local/bin/../share/google/google-cloud-sdk/./lib/googlecloudsdk/surface/components/update.py", line 66, in Run
14:06:04     version=args.version)
14:06:04   File "/usr/local/bin/../share/google/google-cloud-sdk/./lib/googlecloudsdk/core/updater/update_manager.py", line 759, in Update
14:06:04     with install_state.LastUpdateCheck() as update_check:
14:06:04   File "/usr/local/bin/../share/google/google-cloud-sdk/./lib/googlecloudsdk/core/updater/local_state.py", line 232, in LastUpdateCheck
14:06:04     return LastUpdateCheck(self)
14:06:04   File "/usr/local/bin/../share/google/google-cloud-sdk/./lib/googlecloudsdk/core/updater/local_state.py", line 658, in __init__
14:06:04     self._LoadData()
14:06:04   File "/usr/local/bin/../share/google/google-cloud-sdk/./lib/googlecloudsdk/core/updater/local_state.py", line 667, in _LoadData
14:06:04     data = json.loads(fp.read())
14:06:04   File "/usr/lib/python2.7/json/__init__.py", line 326, in loads
14:06:04     return _default_decoder.decode(s)
14:06:04   File "/usr/lib/python2.7/json/decoder.py", line 368, in decode
14:06:04     raise ValueError(errmsg("Extra data", s, end, len(s)))
14:06:04 ValueError: Extra data: line 1 column 102 - line 1 column 103 (char 102 - 103)
14:06:04 
```

**Comment by quinton-hoole** [comment]
Possibly a race while another test run is busy updating the gcloud tools?  Lets try again...

@k8s-bot test this again please

**Comment by brendandburns** [comment]
ok, I added a bail out.  ptal.

thanks!
--brendan

**Comment by brendandburns** [comment]
fwiw, @jlowdermilk is looking into the problems w/ the PR builder.

**Comment by jlowdermilk** [comment]
@k8s-bot test this

**Comment by davidopp** [comment]
reassigning to @quinton-hoole

**Comment by jlowdermilk** [comment]
@k8s-bot test this

File: cluster/gce/list-resources.sh [hunk 2]
```diff
@@ -37,7 +37,21 @@ GREP_REGEX=""
...
+    echo "Attempt ${attempt} to list ${1} in GCE"
```
Comment (ixdy) [inline cluster/gce/list-resources.sh]: use ${resource} instead of ${1}. (same on L47)

File: cluster/gce/list-resources.sh [hunk 3]
```diff
@@ -37,7 +37,21 @@ GREP_REGEX=""
...
+    if result=$(gcloud compute ${resource} list --project=${PROJECT} ${@:2} | grep "${GREP_REGEX}"); then
+      return ${result}
```
Comment (ixdy) [inline cluster/gce/list-resources.sh]: this should be `echo ${result}` (possibly with -e? not sure how newlines are treated in L44) followed by `return`

**Comment by ixdy** [comment]
I've been seeing that gcloud stacktrace on the PR Jenkins for a few days. I'm not sure what's causing it, but it's not the reason this PR is failing.

**Comment by jlowdermilk** [comment]
@ixdy, the gcloud stacktrace was causing the earlier failures but the latest one looks like it's complaining about leaked resources.

**Comment by ixdy** [comment]
@jlowdermilk where do you see this? Leaking resources isn't supposed to fail PR Jenkins test runs.

The [latest run on this PR](https://goto.google.com/prkubekins/job/kubernetes-pull-build-test-e2e-gce/14055/) failed because
```
+ ./cluster/gce/list-resources.sh
./cluster/gce/list-resources.sh: line 45: return: NAME: numeric argument required
```

as I pointed out in a comment on the commit.

**Comment by ixdy** [comment]
A [test run on a different PR](https://goto.google.com/prkubekins/job/kubernetes-pull-build-test-e2e-gce/14054/consoleFull) shows the gcloud stacktrace, but the build succeeded.

**Comment by ixdy** [comment]
ping @brendandburns - seeing this happen more and more.

File: cluster/gce/list-resources.sh (hunk 2 above)
Comment (brendandburns) [inline cluster/gce/list-resources.sh]: done.

File: cluster/gce/list-resources.sh (hunk 3 above)
Comment (brendandburns) [inline cluster/gce/list-resources.sh]: done.

**Comment by brendandburns** [comment]
@ixdy updated, ptal.

thanks
--brendan

**Comment by ixdy** [comment]
lgtm

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please

**Pull request merged**